packages/fluentui/react-northstar/src/components/Datepicker/DatepickerCalendarCell.tsx

```diff
@@ -14,13 +72,113 @@ export const DatepickerCalendarCell = compose<
   'button',
   DatepickerCalendarCellProps,
   DatepickerCalendarCellStylesProps,
-  ButtonProps,
+  {},
   {}
->(Button, {
-  className: datepickerCalendarCellClassName,
-  displayName: 'DatepickerCalendarCell',
-  mapPropsToStylesProps: () => ({
-    text: true,
-    size: 'small',
-  }),
-});
+>(
+  (props, ref, composeOptions) => {
+    const context = useFluentContext();
+    const { setStart, setEnd } = useTelemetry(composeOptions.displayName, context.telemetry);
+    setStart();
+
+    const { className, design, styles, variables, onClick, disabled, selected, label } = props;
+    const unhandledProps = useUnhandledProps(composeOptions.handledProps, props);
+    const ElementType = getElementType(props);
+    const getA11yProps = useAccessibility(props.accessibility, {
+      debugName: composeOptions.displayName,
+      actionHandlers: {
+        performClick: e => {
+          // prevent Spacebar from scrolling
+          e.preventDefault();
+          handleClick(e);
+        },
+      },
+      mapPropsToBehavior: () => ({
+        selected,
+        disabled,
+      }),
+      rtl: context.rtl,
+    });
+
+    const { classes } = useStyles<DatepickerCalendarCellStylesProps>(DatepickerCalendarCell.displayName, {
+      className: composeOptions.className,
+      mapPropsToStyles: () => ({
+        actionable: !!onClick,
+        disabled,
+        selected,
+      }),
+      mapPropsToInlineStyles: () => ({
+        className,
+        design,
+        styles,
+        variables,
+      }),
+      rtl: context.rtl,
+      composeOptions,
+      unstable_props: props,
+    });
+
+    const handleFocus = (e: React.SyntheticEvent) => {
+      _.invoke(props, 'onFocus', e, props);
+    };
+
+    const handleKeyDown = (e: React.SyntheticEvent) => {
+      _.invoke(props, 'onKeyDown', e, props);
+    };
+
+    const handleClick = (e: React.MouseEvent | React.KeyboardEvent) => {
+      if (disabled) {
+        e.preventDefault();
+        return;
+      }
+
+      _.invoke(props, 'onClick', e, props);
+    };
+
+    const element = (
+      <ElementType
+        {...getA11yProps('root', {
+          className: classes.root,
+          onClick: handleClick,
+          onKeyDown: handleKeyDown,
+          onFocus: handleFocus,
+          ref,
+          ...unhandledProps,
+        })}
+      >
+        {label}
+      </ElementType>
+    );
+    setEnd();
+    return element;
+  },
+  {
+    className: datepickerCalendarCellClassName,
+    displayName: 'DatepickerCalendarCell',
+
+    handledProps: [
+      'accessibility',
+      'as',
+      'className',
+      'label',
+      'disabled',
+      'onClick',
+      'onFocus',
+      'selected',
+      'styles',
+      'variables',
+    ],
+  },
+) as ComponentWithAs<'button', DatepickerCalendarCellProps>;
```

Comment:
```suggestion
);
```

No need for a cast in this case 😸